packages/beeq/src/components/date-picker/bq-date-picker.tsx

```diff
@@ -0,0 +1,500 @@
+
+  /**
+   * Indicates whether the Date picker input is disabled or not.
+   * If `true`, the Date picker is disabled and cannot be interacted with.
+   */
+  @Prop({ mutable: true }) disabled?: boolean = false;
+
+  /** If `true`, the clear button won't be displayed */
+  @Prop({ reflect: true }) disableClear? = false;
+
+  /** Represents the distance (gutter or margin) between the Date picker panel and the input element. */
+  @Prop({ reflect: true }) distance?: number = 8;
+
+  /** The ID of the form that the Date picker input belongs to. */
+  @Prop({ reflect: true }) form?: string;
+
+  /** The Date picker input name. */
+  @Prop({ reflect: true }) name!: string;
+
+  /** If `true`, the Date picker panel will be visible. */
+  @Prop({ reflect: true, mutable: true }) open?: boolean = false;
+
+  /** When set, it will override the height of the Date picker panel. */
+  @Prop({ reflect: true, mutable: true }) panelHeight?: string = 'auto';
+
+  /** The Date picker input placeholder text value */
+  @Prop({ reflect: true }) placeholder?: string;
+
+  /** Position of the Date picker panel */
+  @Prop({ reflect: true }) placement?: FloatingUIPlacement = 'bottom-start';
+
+  /** Indicates whether or not the Date picker input is required to be filled out before submitting the form. */
+  @Prop({ reflect: true }) required?: boolean;
+
+  /** Represents the skidding between the Date picker panel and the input element. */
+  @Prop({ reflect: true }) skidding?: number = 0;
+
+  /** Defines the strategy to position the Date picker panel */
+  @Prop({ reflect: true }) strategy?: 'fixed' | 'absolute' = 'fixed';
+
+  /** If `true`, the Date picker panel will accepts more than 1 month to display */
+  @Prop({ reflect: true }) range: boolean = false;
+
+  /** If `true`, the Date picker panel will accepts to select multiple individual dates */
+  @Prop({ reflect: true }) multi: boolean = false;
+
+  /** Number of months to show when range is `true` */
+  @Prop({ reflect: true }) months: number;
+
+  /** A function that takes a date and returns true if the date should not be selectable */
+  @Prop({ reflect: true }) isDateDisallowed?: (date: Date) => boolean;
+
+  /**
+   * The validation status of the Select input.
+   *
+   * @remarks
+   * This property is used to indicate the validation status of the select input. It can be set to one of the following values:
+   * - `'none'`: No validation status is set.
+   * - `'error'`: The input has a validation error.
+   * - `'warning'`: The input has a validation warning.
+   * - `'success'`: The input has passed validation.
+   */
+  @Prop({ reflect: true }) validationStatus: TInputValidation = 'none';
+
+  /** The select input value represents the currently selected date or range and can be used to reset the field to a previous value.
+   * All dates are expected in ISO-8601 format (YYYY-MM-DD). */
+  @Prop({ reflect: true, mutable: true }) value: TInputValue;
```

Comment:
The value prop should be string type only, keep in mind that we're dealing with dates here 😉 

```suggestion
  @Prop({ reflect: true, mutable: true }) value: string;
```